odl/space/fspace.py

```diff
@@ -379,14 +395,6 @@ def __call__(self, x, out=None, **kwargs):
             if ndim == 1:
                 try:
                     out = self._call(x, **kwargs)
-                    if np.ndim(out) == 0 and not scalar_out:
-                        # Don't accept scalar result. A typical situation where
-                        # this occurs is with comparison operators, e.g.
-                        # "return x > 0" which simply gives 'True' for a
-                        # non-empty tuple (in Python 2). We raise TypeError
-                        # to trigger the call with x[0].
```

Comment:
It wasn't even tested for, but i.m.o. we shouldnt interfere with user code. If the user wants to do that test, they should.

Silently failing like this will only be problematic (and stops scalar output from beeing valid, as shown when @ozanoktem tried making a function like `lambda x : 1.0`).